test/src/main/java/org/apache/accumulo/test/functional/MasterMetricsIT.java

```diff
@@ -0,0 +1,372 @@
+  }
+
+  /**
+   * Validate metrics for consistency with in a run cycle.
+   *
+   * @param values
+   *          map of values from one run cycle.
+   */
+  private void sanity(final Map<String,Long> values) {
+
+    assertTrue(values.get("currentFateOps") <= values.get("totalFateOps"));
+
+    long total = values.entrySet().stream().filter(x -> x.getKey().startsWith("FateTxState_"))
+        .mapToLong(Map.Entry::getValue).sum();
+
+    assertTrue(total >= values.get("currentFateOps"));
+  }
+
+  /**
+   * A series of sanity checks for the metrics between different update cycles, some values should
+   * be at least different, and some of the checks can include ordering.
+   *
+   * @param firstSeen
+   *          map of first metric update
+   * @param nextSeen
+   *          map of a later metric update.
+   */
+  private void validate(Map<String,Long> firstSeen, Map<String,Long> nextSeen) {
+    // total fate ops should not decrease.
+    log.debug("Total fate ops.  Before:{}, Update:{}", firstSeen.get("totalFateOps"),
+        nextSeen.get("totalFateOps"));
+    assertTrue(firstSeen.get("totalFateOps") <= nextSeen.get("totalFateOps"));
+  }
+
+  /**
+   * The hadoop metrics file sink published records as a line with comma separated key=value pairs.
+   * This method parses the line and extracts the key, value pair from metrics that start with AccGc
+   * and returns them in a sort map.
+   *
+   * @param line
+   *          a line from the metrics system file sink.
+   * @return a map of the metrics that match REQUIRED_METRICS_KEYS
+   */
+  private Map<String,Long> parseLine(final String line) {
+
+    if (line == null) {
+      return Collections.emptyMap();
+    }
+
+    Map<String,Long> m = new TreeMap<>();
+
+    String[] csvTokens = line.split(",");
+
+    for (String token : csvTokens) {
+      token = token.trim();
+      String[] parts = token.split("=");
+      if (REQUIRED_METRIC_KEYS.contains(parts[0])) {
+        m.put(parts[0], Long.parseLong(parts[1]));
+      }
+    }
+    return m;
+  }
+
+  private boolean lookForExpectedKeys(final Map<String,Long> received) {
+
+    for (String e : REQUIRED_METRIC_KEYS) {
+      if (!received.containsKey(e)) {
+        log.info("Couldn't find {}", e);
+        return false;
+      }
+    }
+
+    return true;
+  }
+
+  /**
+   * FATE metrics only valid when MASTER_FATE_METRICS_ENABLED=true and GENERAL_LEGACY_METRICS=flase
+   *
+   * @return true if test should run
+   */
+  private boolean testDisabled() {
+
+    boolean fateMetricsEnabled =
+        cluster.getSiteConfiguration().getBoolean(Property.MASTER_FATE_METRICS_ENABLED);
+
+    boolean useLegacyMetrics =
+        cluster.getSiteConfiguration().getBoolean(Property.GENERAL_LEGACY_METRICS);
+
+    if (!fateMetricsEnabled || useLegacyMetrics) {
+
+      log.info("master fate metrics are disabled - MASTER_FATE_METRICS_ENABLED={}, "
+          + "GENERAL_LEGACY_METRICS={}", fateMetricsEnabled, useLegacyMetrics);
```

Comment:
Shouldn't this configuration validation be done in MasterMetricsFactory when the properties are read by the Master?  I didn't see any actual check for this.

Reply:
The tests are only valid with hadoop2 metrics - this prevents test failures if legancy metrics were enabled during the testing.  Added a warning if legacy metrics are enabled that additional FATE metrics added by this PR will not be available.